onnxruntime/core/optimizer/propagate_cast_ops.cc

```diff
@@ -1419,29 +1449,22 @@ Status PropagateCastOps::ApplyImpl(Graph& graph, bool& modified, int graph_level
     LOGS(logger, INFO) << "Propagate Cast operations summary:";
     LOGS(logger, INFO) << "Number of passes = " << pass;
     LOGS(logger, INFO) << "Nodes Inserted:";
-    std::for_each(inserted_node_names.begin(), inserted_node_names.end(), [removed_node_names, logger](std::string name) {
-      if (removed_node_names.find(name) == removed_node_names.end()) { LOGS(logger, INFO) << name; } });
+    std::for_each(inserted_nodes.begin(), inserted_nodes.end(), [&](NodeIndex idx) { LOGS(logger, INFO) << graph.GetNode(idx)->Name(); });
 
     LOGS(logger, INFO) << "Nodes Removed:";
-    std::for_each(removed_node_names.begin(), removed_node_names.end(), [logger](std::string name) {
-      if (inserted_node_names.find(name) == inserted_node_names.end()) { LOGS(logger, INFO) << name; } });
+    std::for_each(removed_node_names.begin(), removed_node_names.end(), [&](const std::string& name) { LOGS(logger, INFO) << name; });
 
     LOGS(logger, INFO) << "Nodes Converted to FP16:";
-    std::for_each(converted_node_names.begin(), converted_node_names.end(), [removed_node_names, logger](std::string name) {
-      if (removed_node_names.find(name) == removed_node_names.end() && inserted_node_names.find(name) == inserted_node_names.end()) {
-        LOGS(logger, INFO) << name; } });
+    std::for_each(converted_nodes.begin(), converted_nodes.end(), [&](NodeIndex idx) { LOGS(logger, INFO) << graph.GetNode(idx)->Name(); });
   }
-  inserted_node_names.clear();
-  converted_node_names.clear();
   return Status::OK();
 }
 
 PropagateCastOps::PropagateCastOps(GraphTransformerConfiguration::PropagateCastOpsConfiguration::Strategy strategy,
-                                   size_t level, const std::vector<std::string>& _allow_list,
+                                   size_t level, const std::vector<std::string>& allow_list,
                                    const std::unordered_set<std::string>& compatible_execution_providers) noexcept
-    : GraphTransformer("PropagateCastOps", compatible_execution_providers), level_(level), strategy_(strategy) {
-  fp16_allow_ops[0].clear();  // Remove previously added op types if any.
-  std::copy(_allow_list.begin(), _allow_list.end(), std::inserter(fp16_allow_ops[0], fp16_allow_ops[0].begin()));
+    : GraphTransformer("PropagateCastOps", compatible_execution_providers), level_(level), fp16_allow_ops_(default_fp16_allow_ops), strategy_(strategy) {
+  std::copy(allow_list.begin(), allow_list.end(), std::inserter(fp16_allow_ops_[0], fp16_allow_ops_[0].begin()));
```

Comment:
The user specified ONNX operator names collection is not expected to be arbitrarily large. In fact we expect the number of such operators to be in single digits only. So, this copy should not be concern.